src/main/java/net/snowflake/client/core/BaseSqlInput.java

```diff
@@ -103,4 +114,53 @@ public SQLXML readSQLXML() throws SQLException {
   public RowId readRowId() throws SQLException {
     throw new SnowflakeLoggedFeatureNotSupportedException(session, "readRowId");
   }
+
+  protected <T> T readObjectOfType(Class<T> type, Object value, SQLInput sqlInput)
+      throws SQLException {
+    if (SQLData.class.isAssignableFrom(type)) {
+      SQLData instance = (SQLData) SQLDataCreationHelper.create(type);
+      if (sqlInput == null) {
+        return null;
+      } else {
+        instance.readSQL(sqlInput, null);
+        return (T) instance;
+      }
+    } else if (Map.class.isAssignableFrom(type)) {
+      Object object = value;
+      if (object instanceof JsonSqlInput) {
+        return (T)
+            OBJECT_MAPPER.convertValue(
+                ((JsonSqlInput) object).getInput(), new TypeReference<Map<String, Object>>() {});
+      } else {
+        return (T) ((ArrowSqlInput) object).getInput();
+      }
+    } else if (String.class.isAssignableFrom(type)
+        || Boolean.class.isAssignableFrom(type)
```

Comment:
What about byte array?

Reply:
I don't think so we need support byte arrays by readObject. In this case customer should use readBytes.